### DISCUSSION TITLE
Improved search with filters

### DISCUSSION THREAD

**Discussion by 13hannes11** [organicmaps/organicmaps#397] [Ideas]
I think a great feature to add would be a search with filters

This could include

- Searching for any tags present in the data
- Limiting the search area (to speed up search and prevent irrelevant results)

What do you think?

**Comment by biodranik** [comment]
1. What are your use cases for tags? Can you please provide some examples which also will be useful to other people?
2. Where the current search doesn't work properly in a view area? We did it in a smart way, so it always tries to find the most relevant objects. Of course, it may not work in all cases, so it's good to know these examples.

**Reply by 13hannes11** [reply]
1. One example that I can come up with is searching for a public water dispenser or a fireplace. Generally it allows more flexibility when searching because a wheelchair user could search for places that are accessible.
2. A case where it doesn't work well for me is searching for a street. If you have street names that are the same in different cities its easy to get the wrong results.

**Comment by biodranik** [comment]
1. Does "drinking water" in the search window works for you? (Note: it looks like there is a bug in the map generator, and not every drinking water is found). If it's what you're looking for, then it would be easy to add additional "tags" - we call it "search categories", see [data/categories.txt](https://github.com/organicmaps/organicmaps/tree/master/data/categories.txt) for more details.
2. Can you please provide a concrete example where street search doesn't work? Including your approx current location and where is your current viewport? Also, in the search, you already have an option to "view results on the map", so the search will highlight results in viewport only.

**Reply by 13hannes11** [reply]
On the first note, yes drinking water, works, however sadly other "tags" like fireplace are not easy to find.

Thanks for second. I did not realise that the search also was a lot dependents on the viewport. One example that I have is searching for *Nassauer Straße* while having my viewport over whole of Germany (only having downloaded "Darmstadt" and "Stockholm") and being in Sweden.
Also upon further inspection I realised that adding the city name to the search query makes the result visible quite high up. Therefore, I'd say 2. is actually not an issue.
